src/github.js

```diff
@@ -127,7 +159,241 @@ module.exports = function githubFactory(token, repository) {
+                if(!jiraId && noteData.title){
+                    jiraId = extract(noteData.title, jiraIdExp);
+                }
+                if(!jiraId && noteData.body){
+                    jiraId = extract(noteData.body, jiraIdExp);
+                }
+                if(jiraId){
+                    jiraId = jiraId
+                        .trim()
+                        .replace(/\s/, '-')
+                        .toUpperCase();
+                }
+                return jiraId;
+            };
+
+            if(noteData){
+                const type = extractType();
+                const jiraId = extractJiraId();
+
+                if(type){
+                    note.push(`_${type}_`);
+                }
+                if(jiraId){
+                    note.push(`[${jiraId}](https://oat-sa.atlassian.net/browse/${jiraId})`);
+                }
+                if(note.length){
+                    note.push(':');
+                }
+                if(noteData.title){
+                    note.push(
+                        noteData.title
+                            .replace(typeExp, '')
+                            .replace(jiraIdExp, '')
+                            .replace(/\//g, '')
+                            .replace(/\s\s+/g, ' ')
+                            .trim()
+                    );
+                }
+                if(noteData.number && noteData.url){
+                    note.push(`[#${noteData.number}](${noteData.url})`);
+                }
+
+                if(noteData.user && noteData.assignee){
+                    note.push('(');
+                    note.push(`by [${noteData.user}](https://github.com/${noteData.user})`);
+                    note.push('-');
+                    note.push(`validated by [${noteData.assignee}](https://github.com/${noteData.assignee})`);
+                    note.push(')');
+                }
+            }
+            return note.join(' ');
+        },
+
+        /**
+         * Extract the release notes from a release pull request.
+         * We first retrieve all commits included in the release,
+         * then we filter out to get only sub pull requests.
+         * We retrieve the data from each of this pull request to extract the relevant info.
+         *
+         * @param {String|Number} prNumber - the number of the release pull request
+         * @returns {Promise<String>} resolves with the release note description
+         */
+        extractReleaseNotesFromReleasePR(prNumber) {
+
+            //1. Get all commits from the release PR
+            return this.getPRCommitShas(prNumber)
+                .then( commits => {
+                    if(commits && commits.length){
+                        // we filter out PR inside those commits
+                        // (github considers pr as issues)
+                        return this.searchIssues({
+                            q : `${commits.join('+')}+repo:${repository}+type:pr+base:develop+is:merged`,
```

Comment:
I need to verify if there's a limit in the number of commits you can push in an issue, or the query length. For example if a release PR contains 200 commits...